tests/get_or_create/tests.py

```diff
@@ -484,3 +485,59 @@ def lock_wait():
         updated_person = Person.objects.get(first_name='John')
         self.assertGreater(after_update - before_start, timedelta(seconds=0.5))
         self.assertEqual(updated_person.last_name, 'NotLennon')
+
+
+class InvalidCreateArgumentsTests(SimpleTestCase):
+    msg = "Invalid field name(s) for model Thing: 'nonexistent'."
+
+    def test_get_or_create_with_invalid_defaults(self):
+        with self.assertRaisesMessage(FieldError, self.msg):
+            Thing.objects.get_or_create(name='a', defaults={'nonexistent': 'b'})
+
+    def test_get_or_create_with_invalid_kwargs(self):
+        with self.assertRaisesMessage(FieldError, self.msg):
+            Thing.objects.get_or_create(name='a', nonexistent='b')
+
+    def test_update_or_create_with_invalid_defaults(self):
+        with self.assertRaisesMessage(FieldError, self.msg):
+            Thing.objects.update_or_create(name='a', defaults={'nonexistent': 'b'})
+
+    def test_update_or_create_with_invalid_kwargs(self):
+        with self.assertRaisesMessage(FieldError, self.msg):
+            Thing.objects.update_or_create(name='a', nonexistent='b')
+
+    def test_update_or_create_multiple_invalid_kwargs(self):
+        with self.assertRaisesMessage(FieldError, "Invalid field name(s) for model Thing:") as cm:
+            Thing.objects.update_or_create(name='a', nonexistent='b', invalid='c')
+        self.assertIn('nonexistent', str(cm.exception))
+        self.assertIn('invalid', str(cm.exception))
+
+    def test_update_or_create_multiple_invalid_defaults(self):
```

Comment:
I don't think the test cases need to be duplicated for every case. I was actually more interested to check that the field name joining that adds the comma works fine. You can do `join(sorted(invalid_params))` to make the message deterministic. Go ahead and squash commits too. Thanks!